json-logic/README.md

```diff
@@ -0,0 +1,62 @@
+# CovQuestions Draft
+
+This is the draft of a custom questionaire format for the CovQuestions project.
+
+## Requirements
+
+* Relatively easy to edit
+* Easy to use and implement
+* As platform-independend as possible
+* Describe questions, results, and result-computation logic
+* Questions might be shown or not depending in intermediate results
+* Support for multiple result categories, with their own results and computations
+
+## Considerations
+
+### Result Format
+
+Two possible options: 
+1. "Flat" list of results with a expressione evaluating to a boolean
+2. Result categories with an expression that evaluate to a result id with a list of results per category.
+
+We favor option 2 - with option 1 it is easy to accidentially show two results from the same category. This is impossible with 2 by design at the price of one additional layer of nesting.
+
+### Evaluation
+
+Two possible options: 
+1. Put all variables and questions into the same array and process them in order.
+2. Seperate variables and questions, process all questions in order and (re-)evaluate all variables after each update.
+
+We favor option 2 - it gives the user one less thing to worry about.
+
+#### Structuring Expressions
+
+JSON Logic is powerful, but messy to read/write. There are multiple ways to express the same logic.
+Once we have a better understanding of the task at hand, we should come up with some basic guidelines for
+formatting and structuring.
+
+### Dependencies between variables
+
+We favor a relatively simple logic and no automatic dependency resolution/re-ordering of questions or variables. 
+If things are ordered incorrectly (e.g. the skipIf in a question refers a future question), the questionaire author should discover and repair the issue during testing.
+
+### Answer Representation
+
+For each answered question, we multiple JSON-Logic variables prefixed with the question id:
+
+```
+question_id.value // string, array of string, number, boolean, date, depending on type
+question_id.count // number
+question_id.answered // boolean
+```
+
+### Special Variables
+
+We need to introduce a set of special meta variables:
+* the date when the survey was conducted
+* tbd.
+
+### Working with dates
+
+This is a nasty TODO. 
+A simplified representation would be better for integrating with JSON-Logic, for example using the amount of Days since 1.1.1970 or otherwise unix timestamps.
```

Comment:
:+1:  for timestamps. If we have a parser for simple logic expressions to JSONLogic, we could also introduce functions for dealing with dates.

Reply:
Why don't we leave this to the UI? I think we are overspecific here. The questionnaire only cares about how many days already passed. 

If somebody want to provide a day picker and calculate the days that's fine, but it should be application specific. 

Example: 
A call questionnaire has no way of asking for a date, thus only the time span can be asked. 
Of course they can also ask for the timespan and calculate the day. But this is overly complex (and done twice: ask for timespan -> calculate day -> calculate timespan)

Reply:
Hmm, but how would one deal with absolute days. I could imagine an example where you want to show a question if someone was in a country after a specific date....

Reply:
Yeah, I just wanted to point out that for the current questions we don't have to think about it. 
And that for every question we ask the complexity of the question should be reduced to the minimum. In order to not write overly complex statements in jsonlogic or prejudice the way, the question will be asked by the application. 

Of course, we will have to think about date comparison later on. If questions like this are added.

Reply:
Yep, thats true. Won't be a thing for the first version.

Reply:
>  for the current questions we don't have to think about it

We still have to think a bit about it - this is a classic case where it's quite hard to think about it, but at the same time a bad decision now can make everything hard later.

JSON Logic has no native support for dates, so we will need some sort of trick. 

> he questionnaire only cares about how many days already passed

I think that's not true. @alexanderthieme to the rescue!

Reply:
From Slack: Only days are important, we expose only days for now.